src/core/amount.js

```diff
@@ -614,10 +586,12 @@ function(quality, counterCurrency, counterIssuer, opts) {
     quality as stored :  5 USD          /  3000000 drops
     inverted          :  3000000 drops  /          5 USD
   */
-  var adjusted = opts.inverse ? inverse(value) : value;
-  var nativeAdjusted = adjusted;
+  const valueStr = mantissa.toString() + 'e' + offset.toString();
+  let nativeAdjusted = this._is_native ?
+    new XRPValue(valueStr) : new IOUValue(valueStr);
```

Comment:
But the tests are expecting canonicalization according to XRPValue's canonicalize function when the currency is native. This results in assertion errors where it is expecting the rounded value: 
```AssertionError: '90,442.19667786521/XRP' === '90,442.196677/XRP'```